src/art.rs

```diff
@@ -160,11 +160,11 @@ impl<P: KeyTrait + Clone, V: Clone> Node<P, V> {
     #[inline]
     fn is_full(&self) -> bool {
         match &self.node_type {
-            NodeType::Node1(n) => self.num_children() >= n.size(),
```

Comment:
Going through `self.num_children()` does a match on the `node_type` again. So I changes this to just use the node directly.

Reply:
awesome! Thanks for catching this

Reply:
awesome!